composer/datasets/ade20k.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+import os
+from dataclasses import dataclass
+from math import ceil
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torchvision.transforms.functional as TF
+import yahp as hp
+from PIL import Image
+from torch.utils.data import Dataset
+from torchvision import transforms
+
+from composer.datasets.hparams import DataloaderSpec, DatasetHparams, SyntheticHparamsMixin
+from composer.datasets.synthetic import SyntheticBatchPairDataset
+from composer.utils import ddp
+from composer.utils.data import NormalizationFn, pil_image_collate
+
+
+class RandomResizePair(torch.nn.Module):
+    """Resize the image and target to `base_size` scaled by a randomly sampled value.
+
+    Args:
+        min_scale (float): the minimum value the samples can be rescaled.
+        max_scale (float): the maximum value the samples can be rescaled.
+        base_size (Tuple[int, int]): a specified base size (height x width) to scale to get the resized dimensions.
+            When this is None (default), use the input image size.
+    """
+
+    def __init__(self, min_scale: float, max_scale: float, base_size: Optional[Tuple[int, int]] = None):
+        super().__init__()
+        self.min_scale = min_scale
+        self.max_scale = max_scale
+        self.base_size = base_size
+
+    def forward(self, sample: Tuple[Image.Image, Image.Image]):
+        image, target = sample
+        resize_scale = np.random.random_sample() * (self.max_scale - self.min_scale) + self.min_scale
+        base_height, base_width = self.base_size if self.base_size else (image.height, image.width)
+        resized_dims = (int(base_height * resize_scale), int(base_width * resize_scale))
+        resized_image = TF.resize(image, resized_dims, interpolation=TF.InterpolationMode.BILINEAR)  # type: ignore
+        resized_target = TF.resize(target, resized_dims, interpolation=TF.InterpolationMode.NEAREST)  # type: ignore
+        return resized_image, resized_target
+
+
+class RandomCropPair(torch.nn.Module):
+    """Crop the image and target at a randomly sampled position.
+
+    Args:
+        crop_size (Tuple[int, int]): the size (height x width) of the crop.
+    """
+
+    def __init__(self, crop_size: Tuple[int, int]):
+        super().__init__()
+        self.crop_size = crop_size
+
+    def forward(self, sample: Tuple[Image.Image, Image.Image]):
+        image, target = sample
+        if image.height > self.crop_size[0] or image.width > self.crop_size[1]:
+            i, j, h, w = transforms.RandomCrop.get_params(image, output_size=self.crop_size)  # type: ignore
```

Comment:
can you add comments why these type ignores are required? The error message from pyright will suffice.

Reply:
Yeah these are annoying... Pytorch transforms accept both PIL images and torch Tensors, but the typing only specifies Tensor. Is there a fix here?

Reply:
I added the message `transform typing does not include PIL.Image` does this work?